core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -346,4 +349,23 @@ object DataMirror {
     }
     myItems ++ childItems
   }
+
+  // Function to find the least common ancestor of two nodes
+  private[chisel3] def leastCommonAncestorModule(left: HasId, right: HasId): Option[BaseModule] = {
+    val leftPathSet = modulePath(left, None).toSet
+    val rightPath = modulePath(right, None)
+    rightPath.collectFirst { case p if leftPathSet.contains(p) => p }
+  }
+  // Function to path upwards, stopping if reaching including
+  private[chisel3] def modulePath(h: HasId, until: Option[BaseModule]): Seq[BaseModule] = {
```

Comment:
Mega-nit: could you re-order `def modulePath` to occur before it's usage in the LCA method?